packages/astro/src/assets/build/generate.ts

```diff
@@ -57,6 +57,23 @@ export async function generateImage(
 	const finalFileURL = new URL('.' + filepath, clientRoot);
 	const finalFolderURL = new URL('./', finalFileURL);
 
+	// The original filepath or URL from the image transform
+	const originalImagePath = isLocalImage
+		? (options.src as ImageMetadata).src
+		: (options.src as string);
+
+	if (config.build.assetsRemoveOriginals) {
```

Comment:
This needs to be disabled in SSR, because a SSR page / endpoint might need those assets to do its runtime optimization.